### PR TITLE
Map downloading changes

### PR DESCRIPTION
**Pull request by nukeulater**
Make use of the map downloading UI.
Now host gets map file name using game functions.
Improvements to dedicated server console out by our commands manager.
Re-enabled map file name packet sending.
Fixed bad usage of VirtualProtect.

### CODE REVIEW

File: xlive/H2MOD/Modules/AdvLobbySettings/AdvLobbySettings.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ static void actuallySendPacket() {
 	lobby_settings->set_mp_blind((int)AdvLobbySettings_mp_blind + 1);
 	lobby_settings->set_flashlight((int)AdvLobbySettings_flashlight + 1);
 
-//	network->send_h2mod_packet(teampak);
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/AdvLobbySettings/AdvLobbySettings.cpp:55]: dont we need this now?

File: xlive/H2MOD/Modules/MapManager/MapManager.h [hunk 2]
```diff
@@ -20,6 +20,7 @@ class MapManager {
 	bool hasCustomMap(std::wstring mapName);
 
 	//client functions/data below
+	void gamePatches();
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/MapManager/MapManager.h]: can you rename this, applyGamePatches

File: xlive/H2MOD/GUI/GUI.cpp [hunk 3]
```diff
@@ -654,9 +654,9 @@ int WINAPI XLiveRender()
 			}
 #pragma endregion achievement rendering
 
-			if (GameEngine == 3 && mapManager->getCustomLobbyMessage() != NULL) {
-				drawText(0, 30, COLOR_GOLD, mapManager->getCustomLobbyMessage(), normalSizeFont);
-			}
+			/*if (GameEngine == 3 && mapManager->getCustomLobbyMessage() != NULL) {
```
Comment (shayaantx) [inline xlive/H2MOD/GUI/GUI.cpp]: just remove this if we dont need it anymore

File: xlive/H2MOD/Modules/AdvLobbySettings/AdvLobbySettings.cpp (hunk 1 above)
Comment (shayaantx) [reply xlive/H2MOD/Modules/AdvLobbySettings/AdvLobbySettings.cpp:55]: nvm

File: xlive/H2MOD/Modules/MapManager/MapManager.cpp [hunk 4]
```diff
@@ -45,11 +45,94 @@ std::wstring CUSTOM_MAP = L"Custom Map";
...
+		WriteBytes(h2mod->GetBase() + 0x215A9E, jmp, 1); /* Allow map download in network */
+		WriteBytes(h2mod->GetBase() + 0x215AC9, jmp, 1); /* Disable "Match has begun" bullshit */
+		PatchCall(h2mod->GetBase() + 0x244A4A, display_map_downloading_menu); /* Redirect the menu constructor to our code to replace the game's map downloading code callback */
+
+		// code below is for percentage display
+		PatchCall(h2mod->GetBase() + 0x244B77, get_total_map_downloading_percentage); /* Redirects map downloading percentage to our custom downloader */
+		PatchCall(h2mod->GetBase() + 0x22EE41, get_total_map_downloading_percentage); /* Redirects map downloading percentage to our custom downloader */
+	}
+	else {
+		// DEDICATED SERVER
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/MapManager/MapManager.cpp]: is this necessary? if not, just remove it

File: xlive/H2MOD/Modules/MapManager/MapManager.cpp [hunk 5]
```diff
@@ -195,42 +276,41 @@ const char* MapManager::getCustomLobbyMessage() {
 	return this->customLobbyMessage;
 }
 
+enum game_lobby_states : int
+{
+	not_in_lobby = 0,
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/MapManager/MapManager.cpp]: the first one would already be 0

File: xlive/H2MOD/Modules/MapManager/MapManager.cpp [hunk 6]
```diff
@@ -195,42 +276,41 @@ const char* MapManager::getCustomLobbyMessage() {
...
-			//skip empty map names
-			continue;
-		}
-		std::wstring unicodeMapFilename(mapPath);
-		std::string nonUnicodeCustomMapFilename(unicodeMapFilename.begin(), unicodeMapFilename.end());
-		std::size_t offset = nonUnicodeCustomMapFilename.find_last_of("\\");
-		std::size_t extOffset = nonUnicodeCustomMapFilename.find_last_not_of(mapExt);
-		std::string nonUnicodeMapFileName = nonUnicodeCustomMapFilename.substr(offset + 1, extOffset);
-		if (!nonUnicodeMapFileName.empty() && wcscmp(this->getMapName().c_str(), mapName) == 0) {
+	//typedef int(__cdecl* get_lobby_state)();
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/MapManager/MapManager.cpp]: why you need this?

**Comment by shayaantx** [comment]
can you add some screenshots too? (various states)

**Comment by NukeULater** [comment]
[](https://cdn.discordapp.com/attachments/339567029550055424/526731280281370625/unknown.png)

**Comment by nukeulater** [comment]
Here

**Review by shayaantx — APPROVED** [review]

**Pull request merged**